routes/signUpRoute.js

```diff
@@ -0,0 +1,147 @@
+const User = require('../models/user')
+const router = require('express').Router()
+const jwt = require('jsonwebtoken')
+const bcrypt = require('bcrypt')
+const secretKey = require('../config/secretKey')
+const errorMsg = require('../config/errorMsg')
+const statusCode = require('../config/statusCode')
+const tokenList = {}
+
+router.post('/signup', async(req,res)=>{
+    const user =  await User.findOne({email:req.body.email})
+    if(user) {
+        return res.status(statusCode.forbidden.code).json(errorMsg.user_exists.message)
+    }
+    else{
+        //hash with salting add random string 
+        bcrypt.hash(req.body.password, 10, async(err,hash) =>{
+        if(err){
+            return res.status(statusCode.forbidden.code).json({
+                error:errorMsg.forbidden.message
+            });
+        }
+        else{
+            const token = jwt.sign({name : req.body.name, email: req.body.email}, secretKey.token.key, { expiresIn: '24h' })
+            const refreshToken = await jwt.sign({name : req.body.name, email: req.body.email},secretKey.token.key,{expiresIn: '30d'})
+            try{
+                const user = new User({
+                    name: req.body.name,
+                    email: req.body.email,
+                    password: hash,
+                    refreshToken : refreshToken
+                })
+                user.save()
+                const response = {
+                    "status": "Logged in",
+                    "token": token,
+                    "refreshToken": refreshToken,
```

Comment:
@AwaleRohin you don't need to send all refresh token to the client. just send the token related to that device.